integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala

```diff
@@ -713,16 +714,20 @@ object CommonUtil {
                         SegmentStatusManager.readLoadMetadata(
                           carbonTablePath.getMetadataDirectoryPath)
                       var loadInprogressExist = false
+                      val staleFolders: Seq[CarbonFile] = Seq()
                       listOfLoadFolderDetailsArray.foreach { load =>
                         if (load.getLoadStatus.equals(LoadStatusType.IN_PROGRESS.getMessage) ||
                             load.getLoadStatus.equals(LoadStatusType.INSERT_OVERWRITE.getMessage)) {
                           load.setLoadStatus(CarbonCommonConstants.MARKED_FOR_DELETE)
+                          staleFolders :+ FileFactory.getCarbonFile(
+                            carbonTablePath.getCarbonDataDirectoryPath("0", load.getLoadName))
```

Comment:
Because it is always safe to update the status file first before deleting anything.